x-pack/plugins/maps/public/components/layer_addpanel/view.js

```diff
@@ -4,166 +4,169 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { Component, Fragment } from 'react';
-import { ALL_SOURCES } from '../../shared/layers/sources/all_sources';
+import React, { Component } from 'react';
+import { SourceSelect } from './source_select/source_select';
+import { FlyoutFooter } from './flyout_footer';
+import { SourceEditor } from './source_editor';
+import { ImportEditor } from './import_editor';
 import {
-  EuiButton,
-  EuiButtonEmpty,
   EuiFlexGroup,
-  EuiFlexItem,
   EuiTitle,
-  EuiPanel,
-  EuiSpacer,
-  EuiCard,
-  EuiIcon,
   EuiFlyoutHeader,
-  EuiFlyoutFooter,
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-import _ from 'lodash';
 
 export class AddLayerPanel extends Component {
 
   state = {
     sourceType: null,
-    layer: null
+    layer: null,
+    indexingTriggered: false,
```

Comment:
I think `indexingTriggered`, `indexingSuccess`, `importIndexingReady` state either needs to be in `ImportEditor` or redux store. You are mixing a lot of indexing source logic into `AddLayerPanel`. `AddLayerPanel` should be source agnostic and rely on source state to drive the flow. The `GeojsonFileSource` should drive `AddLayerPanel`, not letting its state bleed into `AddLayerPanel`

Reply:
I never liked the state there too much either. I've moved it to redux instead which will allow us to do some other good UI effects down the road related to import